### DISCUSSION TITLE
Squash commit as default?

### DISCUSSION THREAD

**Discussion by CloseChoice** [shap/shap#3689] [Ideas] [answered]
I messed up merging #3685 and #3677 as I did not squash (since I did not pay attention and thought this is the default). Since it is currently just @connortann and me who are merging I think we can manage this but if the effort to change this is low, I would welcome setting squash merge as default: 

https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests

I cannot see the settings in there, what about you @connortann ?

**Comment by connortann** [comment, marked as answer]
I don't have access to that setting. Unfortunately I believe Scott is the only one who can change that.

However, with just the two of us merging, it shouldn't be too problematic. I think perhaps GitHub might pre-select the most recent merge type. For me, Squash & Merge is pre-selected as the default:

![image](https://github.com/shap/shap/assets/71127464/92dd5ded-b330-4886-83e5-0c3ebb675118)

I don't think a few non-squashed PRs matters particularly. But, let's continue to squash PRs in future to keep things tidy.

**Reply by CloseChoice** [reply]
will do, let's leave it like that

**Reply by connortann** [reply]
PS - now that you have write access to master, we should probably define our typical convention for merging. How about this:
1. When the author is a maintainer, they usually merge their own PRs (after a review)
2. Otherwise, the maintainer who gives a review will also merge

**Reply by CloseChoice** [reply]
totally agree with these conventions

**Reply by connortann** [reply]
FYI @thatlittleboy
